src/Storages/StorageURL.cpp

```diff
@@ -128,6 +128,17 @@ namespace
 
                     try
                     {
+                        std::string userInfo = request_uri.getUserInfo();
```

Comment:
```suggestion
                        std::string user_info = request_uri.getUserInfo();
```

Reply:
Yes, I do. userInfo string is "guest:guest" when input "https://guest:guest@localhost:someport/".
By the way, FunctionalStatelessTestAsan fails(01622_defaults_for_url_engine.sh). I think that's none business with my commit.

Reply:
Actually looks like it must work, are you sure that the problem is in `if(n != std::string::npos)`?

Reply:
userInfo string stores "username:password" string.  Now the condition judge passed the test.
```
try
                    {
                        std::string userInfo = request_uri.getUserInfo();
                        if (!userInfo.empty())
                        {
                            std::size_t n = userInfo.find(":");
                            if(n != std::string::npos)
                            {
                                credentials.setUsername(userInfo.substr(0, n));
                                credentials.setPassword(userInfo.substr(n+1));
                            }
                        }
```

Reply:
> /ClickHouse/src/Storages/StorageURL.cpp:135:                            if(n != std::string::npos)

This is style check, I did not get about what check were you talking about, and thought you solution fails on this if statement check...

You are just missing a space between `if` and `(`

Reply:
Also build check is related:
```
src/CMakeFiles/dbms.dir/Storages/StorageURL.cpp.o.d -o src/CMakeFiles/dbms.dir/Storages/StorageURL.cpp.o -c ../src/Storages/StorageURL.cpp
2021-11-29 07:13:08 /build/obj-x86_64-linux-gnu/../src/Storages/StorageURL.cpp:131:37: error: invalid case style for local variable 'userInfo' [readability-identifier-naming,-warnings-as-errors]
2021-11-29 07:13:08                         std::string userInfo = request_uri.getUserInfo();
2021-11-29 07:13:08                                     ^~~~~~~~
2021-11-29 07:13:08                                     user_info
2021-11-29 07:13:08 /build/obj-x86_64-linux-gnu/../src/Storages/StorageURL.cpp:134:59: error: 'find' called with a string literal consisting of a single character; consider using the more effective overload accepting a character [performance-faster-string-find,-warnings-as-errors]
2021-11-29 07:13:08                             std::size_t n = userInfo.find(":");
2021-11-29 07:13:08                                                           ^~~
2021-11-29 07:13:08                                                           ':'
2021-11-29 07:13:08 91053 warnings generated.
```
Other checks are not related.